E127 on slightly modified PEP-8 approved example (multiline with)

I took the multiline with statement from PEP-8 and tested a few variations. I only get an error when the statement is inside a function and preceded by another statement. This seems related to #316 and #256. git bisect confirms: the first bad commit is 974748fd9a0ecdd494cdc51aeb16c54d38256d58.
## Steps to reproduce

Run pycodestyle on:

```
def f1():
    print('foo')
    with open('/path/to/some/file/you/want/to/read') as file_1, \
         open('/path/to/some/file/being/written', 'w') as file_2:
        file_2.write(file_1.read())


def f2():
    with open('/path/to/some/file/you/want/to/read') as file_1, \
         open('/path/to/some/file/being/written', 'w') as file_2:
        file_2.write(file_1.read())

print('foo')
with open('/path/to/some/file/you/want/to/read') as file_1, \
     open('/path/to/some/file/being/written', 'w') as file_2:
    file_2.write(file_1.read())
```
## Result

`test.py:4:10: E127 continuation line over-indented for visual indent`
## Expected

No error, or at least the same behavior between all three versions. Before 974748fd9a0ecdd494cdc51aeb16c54d38256d58, pycodestyle reported E126 errors for the other two variations.

--- a/pycodestyle.py
+++ b/pycodestyle.py
@@ -695,6 +695,9 @@ def continued_indentation(logical_line, tokens, indent_level, hang_closing,
         elif (token_type in (tokenize.STRING, tokenize.COMMENT) or
               text in ('u', 'ur', 'b', 'br')):
             indent_chances[start[1]] = str
+        # visual indent after assert/raise/with
+        elif not row and not depth and text in ["assert", "raise", "with"]:
+            indent_chances[end[1] + 1] = True
         # special case for the "if" statement because len("if (") == 4
         elif not indent_chances and not row and not depth and text == 'if':
             indent_chances[end[1] + 1] = True

--- a/testsuite/E12.py
+++ b/testsuite/E12.py
@@ -372,4 +372,26 @@ print(True)
 
 print(a
 , end=' ')
-#:
+#: E127:4:12
+def foo():
+    pass
+    raise 123 + \
+           123
+#: E127:4:13
+class Eggs:
+    pass
+    assert 123456 == \
+            123456
+#: E127:4:11
+def f1():
+    print('foo')
+    with open('/path/to/some/file/you/want/to/read') as file_1, \
+          open('/path/to/some/file/being/written', 'w') as file_2:
+        file_2.write(file_1.read())
+#: E127:5:11
+def f1():
+    print('foo')
+    with open('/path/to/some/file/you/want/to/read') as file_1, \
+         open('/path/to/some/file/being/written', 'w') as file_2, \
+          open('later-misindent'):
+        file_2.write(file_1.read())

--- a/testsuite/E12not.py
+++ b/testsuite/E12not.py
@@ -639,3 +639,23 @@ print dedent(
         # more stuff
     )
 )
+
+
+def foo():
+    pass
+    raise 123 + \
+          123
+
+
+class Eggs:
+    pass
+    assert 123456 == \
+           123456
+
+
+def f1():
+    print('foo')
+    with open('/path/to/some/file/you/want/to/read') as file_1, \
+         open('/path/to/some/file/being/written', 'w') as file_2, \
+         open('just-making-sure-more-continuations-also-work'):
+        file_2.write(file_1.read())

